sound/soc/codecs/hdmi-codec.c

```diff
@@ -619,6 +662,20 @@ static const struct snd_soc_dai_ops hdmi_codec_spdif_dai_ops = {
 			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_LE)
 
 struct snd_kcontrol_new hdmi_codec_controls[] = {
+	{
+		.access = SNDRV_CTL_ELEM_ACCESS_READ,
+		.iface = SNDRV_CTL_ELEM_IFACE_MIXER,
```

Comment:
this should be SNDRV_CTL_ELEM_IFACE_PCM like in the IEC958 Playback Default and ELD controls below

Reply:
Hi,
The doc seems to suggest otherwise though, from https://www.kernel.org/doc/html/latest/sound/kernel-api/writing-an-alsa-driver.html#iec958-s-pdif

> “IEC958 Playback Con Mask” is used to return the bit-mask for the IEC958 status bits of consumer mode. Similarly, “IEC958 Playback Pro Mask” returns the bitmask for professional mode. They are read-only controls, and are defined as MIXER controls (iface = SNDRV_CTL_ELEM_IFACE_MIXER).

Reply:
Having mask and default controls on different interfaces would be really odd as these are closely related to each other.

Current kernel code shows mask/default controls are either all defined as mixer controls (eg hda or intel hdmi) or pcm controls (eg ice1712).

see eg `git grep -B 2 "SNDRV_CTL_NAME_IEC958.*PLAYBACK" sound`

Reply:
Some have the doc suggest too (ice1724, oxygen, trident). It's a bit of a mess, I'll try to clear it up while upstreaming :)